google/resource_google_project_default_service_accounts_test.go

```diff
@@ -0,0 +1,287 @@
+					preventRaceCondition(), // Looks like the API is eventually consistent by couple seconds
+					testAccCheckGoogleProjectDefaultServiceAccountsChanges(t, project, action),
+				),
+			},
+		},
+	})
+}
+
+func TestAccResourceGoogleProjectDefaultServiceAccountsDeprivilege(t *testing.T) {
+	t.Parallel()
+
+	org := getTestOrgFromEnv(t)
+	project := fmt.Sprintf("tf-project-%d", randInt(t))
+	billingAccount := getTestBillingAccountFromEnv(t)
+	action := "DEPRIVILEGE"
+	restorePolicy := "REACTIVATE"
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGoogleProjectDefaultServiceAccountsRevert(t, project, action),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckGoogleProjectDefaultServiceAccountsAdvanced(org, project, billingAccount, action, restorePolicy),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("google_project_default_service_accounts.acceptance", "id", "projects/"+project),
+					resource.TestCheckResourceAttrSet("google_project_default_service_accounts.acceptance", "project"),
+					resource.TestCheckResourceAttr("google_project_default_service_accounts.acceptance", "action", action),
+					resource.TestCheckResourceAttrSet("google_project_default_service_accounts.acceptance", "project"),
+					preventRaceCondition(), // Looks like the API is eventually consistent by couple seconds
+					testAccCheckGoogleProjectDefaultServiceAccountsChanges(t, project, action),
+				),
+			},
+		},
+	})
+}
+
+func preventRaceCondition() func(s *terraform.State) error {
+	return func(s *terraform.State) error {
+		time.Sleep(5 * time.Second)
+		return nil
+	}
+}
+
+func testAccCheckGoogleProjectDefaultServiceAccountsAdvanced(org, project, billingAccount, action, restorePolicy string) string {
+	return fmt.Sprintf(`
+resource "google_project" "acceptance" {
+	project_id = "%s"
+	name       = "%s"
+	org_id  = "%s"
+	billing_account = "%s"
+}
+
+resource "google_project_service" "acceptance" {
+	project = google_project.acceptance.project_id
+	service = "compute.googleapis.com"
+
+	disable_dependent_services = true
+}
+
+resource "google_project_default_service_accounts" "acceptance" {
+	depends_on = [google_project_service.acceptance]
+	project = google_project.acceptance.project_id
+	action = "%s"
+	restore_policy = "%s"
+}
+`, project, project, org, billingAccount, action, restorePolicy)
+}
+
+func testAccCheckGoogleProjectDefaultServiceAccountsChanges(t *testing.T, project, action string) func(s *terraform.State) error {
+	return func(s *terraform.State) error {
+		config := googleProviderConfig(t)
+		response, err := config.NewIamClient(config.userAgent).Projects.ServiceAccounts.List(prefixedProject(project)).Do()
+		if err != nil {
+			return fmt.Errorf("failed to list service accounts on project %q: %v", project, err)
+		}
+		for _, sa := range response.Accounts {
+			switch strings.ToLower(sa.DisplayName) {
```

Comment:
Same comment here as in the resource, where I think this switch statement should actually just be an if statement with an || clause